iceberg/pom.xml

```diff
@@ -51,14 +52,24 @@
         <dependencies>
             <dependency>
                 <groupId>org.apache.iceberg</groupId>
-                <artifactId>iceberg-api</artifactId>
+                <artifactId>patched-iceberg-api</artifactId>
                 <version>patched-${iceberg-api.version}-${project.parent.version}</version>
             </dependency>
             <dependency>
                 <groupId>org.apache.iceberg</groupId>
-                <artifactId>iceberg-core</artifactId>
+                <artifactId>patched-iceberg-core</artifactId>
                 <version>patched-${iceberg-api.version}-${project.parent.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.iceberg</groupId>
+                <artifactId>iceberg-api</artifactId>
+                <version>${iceberg-api.version}</version>
```

Comment:
not completely related to this PR, but why isn't this just `iceberg.version`? I guess there shouldn't be different version for iceberg-api, iceberg-data, etc.

Reply:
Hm yeah it should be iceberg.version everywhere, but I was just following suit with this one, as I see the -api version being used in all poms. Maybe we could address this in separate jira.

Reply:
On 2nd thought I'll just go ahead and fix it in the next iteration of this change. I'll keep iceberg-api.version but that will only be applicable to -api lib.